src/typing/type_inference_js.ml

```diff
@@ -3073,12 +3067,20 @@ and expression_ cx loc e = Ast.Expression.(match e with
   | JSXElement e ->
       jsx cx e
 
+  | Class ({ Ast.Class.id } as c) ->
+      let (nloc, name) = (
```

Comment:
Maybe pull this out into a `get_class_name` func (to consolidate with logic up on line 1853)